packages/blueprints-integration/src/documents/part.ts

```diff
@@ -85,6 +86,17 @@ export interface IBlueprintMutatablePart<TPrivateData = unknown, TPublicData = u
 
 	/** MediaObjects that when created/updated, should cause the blueprint to be rerun for the Segment of this Part */
 	hackListenToMediaObjectUpdates?: HackPartMediaObjectSubscription[]
+
+	/** States for UserEdits, could be lock from NRCS updates,
+	 * lock from user changes,
+	 * or removedByUser
+	 * */
+	userEditStates?: Record<string, boolean>
+
+	/**
+	 * User editing definitions for this part
+	 */
+	userEdits?: UserEditingDefinition[]
```

Comment:
we should rename these `userEditOperations?: .....`, to make the name a bit more self descriptive